be/src/olap/rowset/segment_v2/inverted_index_compound_directory.cpp

```diff
@@ -116,7 +116,13 @@ void DorisCompoundFileWriter::sort_files(std::vector<FileInfo>& file_infos) {
 void DorisCompoundFileWriter::writeCompoundFile() {
     // list files in current dir
     std::vector<std::string> files;
-    directory->list(&files);
```

Comment:
suggest a more modular way to impl: encapsulate the ram_dir logic in list/fileLength/... functions, instead of exposing them to caller's code.